apps/web/src/components/Solana/WalletConnectButton.tsx

```diff
@@ -0,0 +1,95 @@
+"use client";
+
+import {
+  Box,
+  Button,
+  Icon,
+  type IconName,
+  Text,
+  type VariantButtonProps,
+} from "@dex-web/ui";
+import { truncate } from "@dex-web/utils";
+import { Popover, PopoverButton, PopoverPanel } from "@headlessui/react";
+import { useWallet } from "@solana/wallet-adapter-react";
+import type { FC, MouseEvent } from "react";
+import { useCallback, useMemo } from "react";
+import { useWalletModal } from "./useWalletModal";
+
+export const WalletConnectButton: FC<VariantButtonProps> = ({
+  onClick,
+  ...props
+}) => {
+  const { visible, setVisible } = useWalletModal();
+
+  const {
+    connected,
+    publicKey,
+    connecting,
+    disconnecting,
+    disconnect,
+    wallet,
+  } = useWallet();
+
+  const buttonText = useMemo(() => {
```

Comment:
Again, does this need useMemo?

The React docs advise 

> You should only rely on useMemo as a performance optimization. If your code doesn’t work without it, find the underlying problem and fix it first. Then you may add useMemo to improve performance.

https://react.dev/reference/react/useMemo